portfolio/src/main/webapp/script.js

```diff
@@ -28,15 +31,59 @@ function addRandomFunFacts() {
   factsContainer.innerText = facts;
 }
 
+// adds a collapsible button 
 function ReadMore() {
   var moreText = document.getElementById("more");
   var btnText = document.getElementById("readmorebtn");
 
-  if (more.style.display === "inline") {
+   if (moreText.style.display === "inline") {
     btnText.innerHTML = "+ READ MORE"; 
     moreText.style.display = "none";
   } else {
-    btnText.innerHTML = "- READ LESS"; 
+    btnText.innerHTML = "- READ LESS";
+    moreText.style.display = "inline";
+    
+  }
+}
+
+function ReadMore1() {
```

Comment:
All these methods do the same thing. Instead of defining new methods. you can do

function ReadMore(moreId, buttonText) {
  var moreText = document.getElementById(moreId);
  var buttonText = document.getElementById(buttonText);

   if (moreText.style.display === "inline") {
    buttonText.innerHTML = "+ READ MORE"; 
    moreText.style.display = "none";
  } else {
    buttonText.innerHTML = "- READ LESS";
    moreText.style.display = "inline";

  }
}

Then call these using:
ReadMore("more", "readmorebtn")
ReadMore("more1", "readmorebtn1")
ReadMore("more2", "readmorebtn2"
ReadMore("more3", "readmorebtn3")

Reply:
I meant function ReadMore(moreId, buttonId)

Reply:
Can you also add a comment on what the function does for every function you've defined so far.

Reply:
thank you for the advice on the functions! very helpful!
And yep ill add the comments